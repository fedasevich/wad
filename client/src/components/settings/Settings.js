import { observer } from 'mobx-react-lite';
import React from 'react';
import { Col, Row } from 'react-bootstrap';
import MenuProvider from '../MenuProvider';
import { SettingGroup } from './SettingGroup';
import { CloseCalculatorOnSubmit } from './closeCalculatorOnSubmit/CloseCalculatorOnSubmit';
import { PersistTransactionFilter } from './persistTransactionFilter/PersistTransactionFilter';
import { StartPageSettings } from './startPage/StartPageSettings';
import { Theme } from './theme/Theme';


export const Settings = observer(({ }) => {
    return (
        <Row className='vh-80'>
            <Col md={{ offset: 1, span: 10 }} >
                <MenuProvider>
                    <MenuProvider.Container >
                        <SettingGroup name={'Start page'} >
                            <StartPageSettings />
                        </SettingGroup>
                    </MenuProvider.Container>
                </MenuProvider>
                <MenuProvider>
                    <MenuProvider.Container >
                        <SettingGroup name={'Calculator'} >
                            <CloseCalculatorOnSubmit />
                        </SettingGroup>
                    </MenuProvider.Container>
                </MenuProvider>
                <MenuProvider>
                    <MenuProvider.Container >
                        <SettingGroup name={'Transaction'} >
                            <PersistTransactionFilter />
                        </SettingGroup>
                    </MenuProvider.Container>
                </MenuProvider>
                <MenuProvider>
                    <MenuProvider.Container >
                        <SettingGroup name={'Theme'} >
                            <Theme />
                        </SettingGroup>
                    </MenuProvider.Container>
                </MenuProvider>
            </Col>
            {/* <Col md={{ offset: 1, span: 4 }} >
                <MenuProvider>
                    <MenuProvider.Container>
                    </MenuProvider.Container>
                </MenuProvider>
            </Col> */}
        </Row>
    );
})