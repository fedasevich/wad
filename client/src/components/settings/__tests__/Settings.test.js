import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Settings } from '../Settings';

describe('Settings.js', () => {
  it('renders without errors', () => {
    const { container } = render(<Settings />);
    expect(container).toMatchSnapshot();
  });
});
