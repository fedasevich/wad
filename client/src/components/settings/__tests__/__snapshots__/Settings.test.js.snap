// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings.js renders without errors 1`] = `
<div>
  <div
    class="vh-80 row"
  >
    <div
      class="col-md-10 offset-md-1"
    >
      <div
        class=" menu mb-3"
      >
        <div
          class="menu-container d-flex flex-column "
        >
          <div
            class="mb-3"
          >
            <h3
              class="mb-2 fw-medium"
            >
              Currency
            </h3>
            <select
              class="form-select"
            >
              <option
                value="1"
              >
                United States Dolar
                 - 
                $
              </option>
              <option
                value="2"
              >
                Euro
                 - 
                €
              </option>
              <option
                value="3"
              >
                Czech koruna
                 - 
                Kč
              </option>
              <option
                value="4"
              >
                Ukrainian hrivnia
                 - 
                ₴
              </option>
              <option
                value="5"
              >
                Polish złoty
                 - 
                zł
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <h3
              class="mb-2 fw-medium"
            >
              Start page
            </h3>
            <select
              class="form-select"
            >
              <option
                value="/main"
              >
                /main
              </option>
              <option
                value="/category"
              >
                /category
              </option>
              <option
                value="/analytics"
              >
                /analytics
              </option>
              <option
                value="/settings"
              >
                /settings
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <h3
              class="mb-2 fw-medium"
            >
              Calculator
            </h3>
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="closeCalculatorOnSubmit"
                name="closeCalculatorOnSubmit"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="closeCalculatorOnSubmit"
                title=""
              >
                Close calculator on submit
              </label>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <h3
              class="mb-2 fw-medium"
            >
              Transaction
            </h3>
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="persistTransactionFilter"
                name="persistTransactionFilter"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="persistTransactionFilter"
                title=""
              >
                Persist transaction filter
              </label>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <h3
              class="mb-2 fw-medium"
            >
              Theme
            </h3>
            <select
              class="form-select"
            >
              <option
                value="default"
              >
                default
              </option>
              <option
                value="dark"
              >
                dark
              </option>
              <option
                value="light"
              >
                light
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
